test/acceptance/tests/connect/connect_inject_test.go

```diff
@@ -50,6 +50,25 @@ func TestConnectInject(t *testing.T) {
 
 			consulCluster.Create(t)
 
+			consulClient := consulCluster.SetupConsulClient(t, c.secure)
+
+			// Check that the ACL token is deleted.
+			if c.secure {
+				// We need to register the cleanup function before we create the deployments
+				// because golang will execute them in reverse order i.e. the last registered
+				// cleanup function will be executed first.
+				t.Cleanup(func() {
+					retry.Run(t, func(r *retry.R) {
+						tokens, _, err := consulClient.ACL().TokenList(nil)
+						require.NoError(r, err)
+						for _, token := range tokens {
+							require.NotContains(r, token.Description, staticServerName)
+							require.NotContains(r, token.Description, staticClientName)
+						}
```

Comment:
Probably a super nit comment but I'm wondering if this should exit if `TokenList()` returns error (during shutdown or otherwise), or it may end up stuck in a retry loop?

Reply:
It won't get stuck in retry loop since the retry function times out after 7 seconds by default. I think it's ok in this case since it won't add that much time to the test running time.